server/utils/assessmentUtils.ts

```diff
@@ -196,13 +196,36 @@ export const statusChangeMessage = (assessmentId: string, status: AssessmentStat
   }
 }
 
-const isUserNote = (note: Note): note is UserNote => {
+const isUserNote = (note: UserNote | SystemNote): note is UserNote => {
   return note.type === 'user'
 }
 
-const userNoteHtml = (note: UserNote) => formatLines(note.message)
+const isSystemNoteWithDetails = (note: UserNote | SystemNote): note is ReferralHistorySystemNote => {
+  return Boolean(note.type === 'system' && note.messageDetails)
+}
+
+export const renderSystemNote = (note: ReferralHistorySystemNote): TimelineItem['html'] => {
+  const reason = note.messageDetails.rejectionReasonDetails || note.messageDetails.rejectionReason
+  const isWithdrawn = note.messageDetails.isWithdrawn ? 'Yes' : 'No'
+
+  const lines = [`Rejection reason: ${reason}`, `Withdrawal requested by the probation practitioner: ${isWithdrawn}`]
+
+  return formatLines(lines.join('\n\n'))
+}
+
+export const renderNote = (note: UserNote | SystemNote): TimelineItem['html'] => {
+  if (isSystemNoteWithDetails(note)) {
```

Comment:
Whats the difference between a systemNote and a userNote?

Reply:
A `UserNote` is a note a user has added manually, which always contains a `message` property. A `SystemNote` is one generated by the system at any point of the process: status changes, mostly, but also original submission, etc. The back-end models these on the same base model (I think!), but now `SystemNote` also has a `messageDetails` property which we can use to render more dynamic content, as is the case with this, to add details about the rejection.